web/api/v1/api_remote_write.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2016 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/prometheus/prometheus/storage/remote"
+)
+
+// QueueRetriever provides a list of remote write queues
+type QueueRetriever interface {
+	Queues() []*remote.QueueManager
+}
+
+// RemoteWritesDiscovery exposes all of the configured remote writes and their status
+type RemoteWritesDiscovery struct {
+	Queues []*RemoteWriteQueue `json:"queues"`
+}
+
+// RemoteWrite represents a single configured remote write and its status
+type RemoteWriteQueue struct {
+	Name     string `json:"name"`
+	Endpoint string `json:"endpoint"`
+
+	ShardsMax     int `json:"shardsMax"`
+	ShardsMin     int `json:"shardsMin"`
+	ShardsCurrent int `json:"shardsCurrent"`
+
+	ShardingCalculations RemoteWriteShardingCalculations `json:"shardingCalculations"`
+	IsResharding         bool                            `json:"isResharding"`
+
+	Shards []*RemoteWriteShard `json:"shards"`
+}
+
+// RemoteWriteShardingCalculations provides insight into the values that are used to calculate the required shard num
+type RemoteWriteShardingCalculations struct {
```

Comment:
Rather than creating a new struct, could this just use `remote.ShardingCalculations`?

Reply:
Yeah, I wasn't too sure if we tried to keep all API objects defined in api/v1, I did consider just adding the json tags to the remote.ShardingCalculations

Reply:
We've a mix currently. We've had problems in cases where the actual struct didn't match what we wanted on the wire, but if that's not the case this should be okay to reuse. This API will be marked experimental one way or the other.

Reply:
Perfect, thanks for clearing that up @brian-brazil